keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -0,0 +1,169 @@
+package com.netflix.spinnaker.keel.services
+
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.core.api.ArtifactSummary
+import com.netflix.spinnaker.keel.core.api.ArtifactSummaryInEnvironment
+import com.netflix.spinnaker.keel.core.api.ArtifactVersionSummary
+import com.netflix.spinnaker.keel.core.api.ArtifactVersions
+import com.netflix.spinnaker.keel.core.api.EnvironmentSummary
+import com.netflix.spinnaker.keel.pause.ActuationPauser
+import com.netflix.spinnaker.keel.persistence.ArtifactRepository
+import com.netflix.spinnaker.keel.persistence.KeelRepository
+import com.netflix.spinnaker.keel.persistence.ResourceStatus
+import com.netflix.spinnaker.keel.persistence.ResourceSummary
+import com.netflix.spinnaker.kork.web.exceptions.InvalidRequestException
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+/**
+ * Service object that offers high-level APIs for application-related operations.
+ */
+@Component
+class ApplicationService(
+  private val pauser: ActuationPauser,
+  private val repository: KeelRepository,
+  private val artifactRepository: ArtifactRepository
+) {
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  fun hasManagedResources(application: String) = repository.hasManagedResources(application)
+
+  /**
+   * Returns a list of [ResourceSummary] for the specified application.
+   *
+   * This function assumes there's a single delivery config associated with the application.
+   */
+
+  fun getResourceSummariesFor(application: String): List<ResourceSummary> {
+    var resources = repository.getSummaryByApplication(application)
+    resources = resources.map { summary ->
+      if (pauser.resourceIsPaused(summary.id)) {
+        // we only update the status if the individual resource is paused,
+        // because the application pause is reflected in the response as a top level key.
+        summary.copy(status = ResourceStatus.PAUSED)
+      } else {
+        summary
+      }
+    }
+    // val constraintStates = repository.constraintStateFor(application)
```

Comment:
why the commented out line here?

Reply:
We had a `constraintsSomething` field at the top level of the controller response earlier, which is where I think this came from IIRC, but upon discussing with @erikmunson we agreed that constraints will show up elsewhere. AFAIK the old field was always returning empty anyway.